go-controller/pkg/ovn/address_set/address_set.go

```diff
@@ -554,63 +562,63 @@ func (as *ovnAddressSet) getIPs() ([]string, error) {
 	return addrset.Addresses, nil
 }
 
-// addIPs appends the set of IPs to the existing address_set.
-func (as *ovnAddressSet) addIPs(ips []net.IP) ([]ovsdb.Operation, error) {
-	if len(ips) == 0 {
+// addAddresses appends the set of addresses to the existing address_set.
+func (as *ovnAddressSet) addAddresses(addresses []string) ([]ovsdb.Operation, error) {
+	if len(addresses) == 0 {
 		return nil, nil
 	}
 
-	uniqIPs := ipsToStringUnique(ips)
+	uniqAddresses := getUniqueAddresses(addresses)
 
-	if as.hasIPs(uniqIPs...) {
+	if as.hasAddresses(uniqAddresses...) {
 		return nil, nil
 	}
 
-	klog.V(5).Infof("(%s) adding IPs (%s) to address set", asDetail(as), uniqIPs)
+	klog.V(5).Infof("(%s) adding Addresses (%s) to address set", asDetail(as), uniqAddresses)
 
 	addrset := nbdb.AddressSet{
 		UUID: as.uuid,
 		Name: as.hashName,
 	}
-	ops, err := libovsdbops.AddIPsToAddressSetOps(as.nbClient, nil, &addrset, uniqIPs...)
+	ops, err := libovsdbops.AddAddressesToAddressSetOps(as.nbClient, nil, &addrset, uniqAddresses...)
 	if err != nil {
-		return nil, fmt.Errorf("failed to add IPs %v to address set %+v: %v", uniqIPs, addrset, err)
+		return nil, fmt.Errorf("failed to add addresses %v to address set %+v: %v", uniqAddresses, addrset, err)
 	}
 
 	return ops, nil
 }
 
-// hasIPs returns true if an address set contains all given IPs
-func (as *ovnAddressSet) hasIPs(ips ...string) bool {
-	existingIPs, err := as.getIPs()
+// hasAddresses returns true if an address set contains all given Addresses
+func (as *ovnAddressSet) hasAddresses(addresses ...string) bool {
+	existingAddresses, err := as.getAddresses()
 	if err != nil {
 		return false
 	}
 
-	if len(existingIPs) == 0 {
+	if len(existingAddresses) == 0 {
 		return false
 	}
 
-	return sets.NewString(existingIPs...).HasAll(ips...)
+	return sets.NewString(existingAddresses...).HasAll(addresses...)
 }
 
-// deleteIPs removes selected IPs from the existing address_set
-func (as *ovnAddressSet) deleteIPs(ips []net.IP) ([]ovsdb.Operation, error) {
-	if len(ips) == 0 {
+// deleteAddresses removes selected addresses from the existing address_set
+func (as *ovnAddressSet) deleteAddresses(addresses []string) ([]ovsdb.Operation, error) {
```

Comment:
A quick question:  how does this work if we call addAddresses with the same IP multiple times and then deleteAddresses once for the same IP?  My understanding is that the address will be removed from the database in the end.  How does that work if the same address comes from multiple sources?  Or is it not possible to get the same IP from multiple places (e.g. once from ANP and once from a CIDR block) ?

Reply:
The reason I ask is primarily that the code is written as if it expects potentially non-unique addresses.  That might be a bit misleading for a reader.  And might be dangerous if they are in fact non-unique.

Reply:
we filter them out - for a given address-set we do not pass in duplicate ips, we ensure they are unique before we send the transact (both during add and delete)

for most of the features address-sets are tied to namespaces and podIPs are unique we are not going to be adding multiple duplicated ips into the address-set where they hold different meanings. So if you delete them from one spot its meant to go away from the database which is the expected behaviour.

Reply:
Ack.  I just had an impression from discussions on this PR that you're going to mix IPs from different sources in the same address set and that made me think about possible duplicates.  But if that is not the case and user always sets unique values for the list of CIDRs, that should be fine.

Reply:
oh, what??? did I edit your comment? I thought I quote replied.. dangit
sorry @igsilya ! I am unsure how to fix that..

Reply:
> Ack. I just had an impression from discussions on this PR that you're going to mix IPs from different sources in the same address set and that made me think about possible duplicates. 

the PR changes an addressset PR from being "ONLY contains IPs" to "contains both IPs and CIDRs" rest of the logic is not changing

> But if that is not the case and user always sets unique values for the list of CIDRs, that should be fine.

there is API validation in place to ensure CIDRs for same peer contain unique elements that is a set of cidrs; in addition to that we do the unique filtering on ovnk side before we add transact

Reply:
That's fine. :)  The repo owners / org members are able to edit comments from others on github, IIRC.  The history is seen in the 'edited by ...' dropdown.

Anyways, thanks for explanation!